cypress/e2e/GUI/ui-funcionality-tests.cy.js

```diff
@@ -0,0 +1,155 @@
+describe('EngageSphere', () => {
+  beforeEach(() => {
+    Cypress.on('window:before:load', window => {
+      window.document.cookie = 'cookieConsent=accepted'
+    })
+    cy.visit('/');
+  });
+
+  it('Should maintain filters when returning from customer details', () => {
+    cy.get('[data-testid="size-filter"]')
+      .select('Small');
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+      cy.contains('button', 'Back')
+      .click();
+    cy.get('[data-testid="size-filter"]')
+      .should('have.value', 'Small');
+  });
+
+  it('Should return to the customer list after clicking back', () => {
+    cy.contains('button', 'View')
+      .click()
+    cy.get('[class^="CustomerDetail"] h2')
+      .should('be.visible');
+    cy.contains('button', 'Back')
+      .click();
+    cy.get('[data-testid="table"]')
+      .should('be.visible');
+  });
+
+  it('Should display the footer with the correct text and links', () => {
+    cy.contains('p', 'Copyright 2024 - Talking About Testing')
+    .should('be.visible');
```

Comment:
Atenção com a indentação.

Aqui devia ser assim:

```suggestion
      .should('be.visible');
```

Reply:
O mesmo vale para outros lugares onde a indentação está incorreta.

Reply:
Consistência é importante para um bom design.